internal/app/upstream/client.go

```diff
@@ -175,7 +179,12 @@ func (m *client) handleStreamsWithRetry(
 	)
 	for {
 		if m.timeout != 0 {
-			childCtx, cancel = context.WithTimeout(ctx, time.Duration(m.timeout)*time.Second)
+			timeout := m.timeout
+			if m.jitter > 0 && m.jitter < timeout {
+				timeout = m.timeout - rand.Int63n(m.jitter)
+			}
+			m.logger.With("aggregated_key", aggregatedKey).Debug(ctx, "Connecting to upstream with timeout: %ds", timeout)
+			childCtx, cancel = context.WithTimeout(ctx, time.Duration(timeout)*time.Second)
 		} else {
 			childCtx, cancel = context.WithCancel(ctx)
```

Comment:
Could you also add a log line here that says connecting to upstream? Just to ensure we have parity since you add the message with the timeout above.

Reply:
done